pinot-segment-local/src/main/java/org/apache/pinot/segment/local/recordtransformer/DataTypeTransformer.java

```diff
@@ -40,14 +47,24 @@
  */
 @SuppressWarnings("rawtypes")
 public class DataTypeTransformer implements RecordTransformer {
+  private static final Logger LOGGER = LoggerFactory.getLogger(DataTypeTransformer.class);
+
   private final Map<String, PinotDataType> _dataTypes = new HashMap<>();
+  private boolean _useDefaultValueOnError;
```

Comment:
Let's make 2 configs for these 2 things: `continueOnError` and `validateTimeValue`

Suggest changing `useDefaultValueOnError` to `continueOnError` because it can be used for other ingestion errors, and we don't want to tie it to the implementation

Reply:
done